source/types.ts

```diff
@@ -0,0 +1,233 @@
+
+	@default 'json'
+	*/
+	fileExtension?: string;
+
+	/**
+	The config is cleared if reading the config file causes a `SyntaxError`. This is a good default, as the config file is not intended to be hand-edited, so it usually means the config is corrupt and there's nothing the user can do about it anyway. However, if you let the user edit the config file directly, mistakes might happen and it could be more useful to throw an error when the config is invalid instead of clearing. Disabling this option will make it throw a `SyntaxError` on invalid config instead of clearing.
+
+	@default true
+	*/
+	clearInvalidConfig?: boolean;
+
+	/**
+	Function to serialize the config object to a UTF-8 string when writing the config file.
+
+	You would usually not need this, but it could be useful if you want to use a format other than JSON.
+
+	@default value => JSON.stringify(value, null, '\t')
+	*/
+	readonly serialize?: Serialize<T>;
+
+	/**
+	Function to deserialize the config object from a UTF-8 string when reading the config file.
+
+	You would usually not need this, but it could be useful if you want to use a format other than JSON.
+
+	@default JSON.parse
+	*/
+	readonly deserialize?: Deserialize<T>;
+
+	/**
+	__You most likely don't need this. Please don't use it unless you really have to.__
+
+	Suffix appended to `projectName` during config file creation to avoid name conflicts with native apps.
+
+	You can pass an empty string to remove the suffix.
+
+	For example, on macOS, the config file will be stored in the `~/Library/Preferences/foo-nodejs` directory, where `foo` is the `projectName`.
+
+	@default 'nodejs'
+	*/
+	readonly projectSuffix?: string;
+
+	/**
+	Access nested properties by dot notation.
+
+	@default true
+
+	@example
+	```
+	const config = new Conf();
+
+	config.set({
+		foo: {
+			bar: {
+				foobar: '🦄'
+			}
+		}
+	});
+
+	console.log(config.get('foo.bar.foobar'));
+	//=> '🦄'
+	```
+
+	Alternatively, you can set this option to `false` so the whole string would be treated as one key.
+
+	@example
+	```
+	const config = new Conf({accessPropertiesByDotNotation: false});
+
+	config.set({
+		`foo.bar.foobar`: '🦄'
+	});
+
+	console.log(config.get('foo.bar.foobar'));
+	//=> '🦄'
+	```
+
+	*/
+	readonly accessPropertiesByDotNotation?: boolean;
+
+	/**
+	Watch for any changes in the config file and call the callback for `onDidChange` if set. This is useful if there are multiple processes changing the same config file.
+
+	__Currently this option doesn't work on Node.js 8 on macOS.__
+
+	@default false
+	*/
+	readonly watch?: boolean;
+}
+
+export type Migrations<T> = {
+	[semverSpecifier: string]: (store: Conf<T>) => void;
+};
+
+export type ConfSchema<T> = { [Property in keyof T]: ValueSchema };
```

Comment:
This is the only type with `Conf` prefix, so I think we should drop it.
```suggestion
export type Schema<T> = {[Property in keyof T]: ValueSchema};
```